### PR TITLE
Add more documentation for TBC and related RPC protocol

### PR DESCRIPTION
**Pull request by marcopeereboom**
**Summary**

**Changes**

### CODE REVIEW

**Review by ClaytonNorthey92 — APPROVED** [review]

File: api/protocol/README.md
Comment (jsorkin24) [reply api/protocol/README.md:27]: Can you explain what you mean?

File: api/protocol/README.md
Comment (jsorkin24) [reply api/protocol/README.md:26]: agreed!

File: api/protocol/README.md
Comment (jsorkin24) [reply api/protocol/README.md:10]: agreed!

File: api/protocol/README.md
Comment (jsorkin24) [reply api/protocol/README.md:7]: will remove!

File: api/protocol/README.md
Comment (jsorkin24) [reply api/protocol/README.md:39]: agreed but which ones should they be in so I know moving forward?

File: api/protocol/README.md
Comment (jsorkin24) [reply api/protocol/README.md:30]: hmm going to try again to see if it works, maybe its the indentation

File: api/protocol/README.md
Comment (joshuasing) [reply api/protocol/README.md]: I think it would be best to have the license notice in just the root project README.md, as it applies to all code in the repository.

File: api/protocol/README.md
Comment (joshuasing) [reply api/protocol/README.md]: e.g.
- **`command`** is the name of the command being executed.
- **`id`** is a string used to uniquely identify each request. ...

File: README.md
Comment (joshuasing) [inline README.md:75]: All commands in this file are shell commands and are not specific to the Bash shell, I think `shell` would be better for formatting.

File: cmd/tbcd/README.md
Comment (joshuasing) [inline cmd/tbcd/README.md:93]:
```suggestion
When the `TBC_ADDRESS` is set to an address (e.g., `localhost:8082`), the`tbcd` daemon listens on the provided address.
```

File: cmd/tbcd/README.md
Comment (joshuasing) [inline cmd/tbcd/README.md:103]: I think the license notice is only needed in the root project README.md.

File: api/protocol/README.md
Comment (jsorkin24) [reply api/protocol/README.md]: I think this structure is better, more so like a definition

File: README.md
Comment (jsorkin24) [reply README.md:89]: sounds good!

File: README.md
Comment (jsorkin24) [reply README.md:75]: fixed

File: README.md
Comment (jsorkin24) [reply README.md:9]: okay will remove

File: README.md
Comment (jsorkin24) [reply README.md:4]: removed

File: README.md
Comment (jsorkin24) [reply README.md:49]: fixed

File: README.md
Comment (jsorkin24) [reply README.md:107]: agreed!

File: README.md
Comment (jsorkin24) [reply README.md:112]: fair!

File: README.md
Comment (jcvernaleo) [inline README.md:37]: Is that grammatically reasonable?  I'd go with `The Hemi network is compriseed of three key services` or something more like that.

File: README.md
Comment (jcvernaleo) [inline README.md:74]: Should we require 1.22.2+ here since that is what we build with?

File: README.md
Comment (jcvernaleo) [inline README.md:98]: Is there a reason to use the less generic `bash` over `shell` for syntax?  Not saying we have to change back but I have a slight preference for `shell` myself.

Comment (jcvernaleo) [reply README.md:98]: That said, if we are being consistent and using `bash` everywhere we should stick with that.  Consistency over personal preferences :)

File: api/protocol/README.md
Comment (jcvernaleo) [inline api/protocol/README.md:29]: Do we actually need this point here?  As long as our rpc doesn't care, it really is up to the user.  I frequently use non-random json ids to identify what a respond is and not just to match up with the initial request (when stress testing for example).  Seems like it should be up to the user and we shouldn't say anything here.

File: cmd/tbcd/README.md
Comment (jcvernaleo) [inline cmd/tbcd/README.md:20]: We probably should fill something in rather than having a XXX note in docs.

File: cmd/tbcd/README.md
Comment (jcvernaleo) [inline cmd/tbcd/README.md:28]: Same comment/question as above on the go version.

File: cmd/tbcd/README.md
Comment (jcvernaleo) [inline cmd/tbcd/README.md:40]: This should match the other code blocks (either all shell or all bash).

File: cmd/tbcd/README.md
Comment (jcvernaleo) [inline cmd/tbcd/README.md:51]: same

File: cmd/tbcd/README.md
Comment (jcvernaleo) [inline cmd/tbcd/README.md:85]: since tbcd is written to `bin/` by defualt we can make this:
```
./bin/tbcd --help
```
I think.

**Review by jcvernaleo — CHANGES_REQUESTED** [review]
Bunch of minor comments.

File: api/protocol/README.md
Comment (joshuasing) [reply api/protocol/README.md:29]: I think I mentioned this as more of a recommendation than a requirement.
I think most users will be able to figure out that it should be unique when creating an API client, however, so I think removing this would be fine. Do you think we should remove it?

Comment (jcvernaleo) [reply api/protocol/README.md:29]: I'd remove in the spirit of not having the keeping the docs a little leaner.

File: cmd/tbcd/README.md
Comment (joshuasing) [reply cmd/tbcd/README.md]: I think that's better

File: cmd/tbcd/README.md
Comment (joshuasing) [reply cmd/tbcd/README.md]: Yep, updated.

File: README.md
Comment (joshuasing) [reply README.md]: `shell` should be used everywhere. I believe Jane changed it to use `bash` in most places for some reason. 
I have reverted, everything should be `shell` once again - thanks!

File: README.md
Comment (joshuasing) [reply README.md]: Yes, agreed.

File: README.md
Comment (joshuasing) [reply README.md]: "is comprised of" seems to be less correct, I have reworded it to use "consists of" - is that okay?

File: api/protocol/README.md
Comment (joshuasing) [reply api/protocol/README.md]: Okay, removed

File: cmd/tbcd/README.md
Comment (joshuasing) [reply cmd/tbcd/README.md:20]: What do you mean by 'fill something in'? 🤔 
I believe this was an estimation from when it was being run as a test for mainnet, so this amount is likely going to change quite a bit before we have a rough number.

File: README.md
Comment (jcvernaleo) [reply README.md]: Yes, better than mine.

File: cmd/tbcd/README.md
Comment (jcvernaleo) [reply cmd/tbcd/README.md:20]: I pretty much just meant pick a number and go with it since nothing we put in there will be correct for too long anyway

**Review by jcvernaleo — APPROVED** [review]
Looks great now.

OK

**Review by joshuasing — APPROVED** [review]

**Comment by joshuasing** [comment]
Waiting for reviews from @ClaytonNorthey92 and if possible @jsorkin24 before merging.

**Review by ClaytonNorthey92 — APPROVED** [review]

**Review by jsorkin24 — APPROVED** [review]
Looks great!

**Pull request merged**
